### PR TITLE
Migrate node (cli, executor, primitives and runtime) to the 2018 edition

### PR DESCRIPTION
**Pull request by DarkEld3r**
Part of #1415.

### CODE REVIEW

File: node/runtime/src/lib.rs [hunk 1]
```diff
@@ -20,54 +20,28 @@
...
-extern crate srml_timestamp as timestamp;
-extern crate srml_treasury as treasury;
-#[macro_use]
-extern crate sr_version as version;
-extern crate node_primitives;
-extern crate substrate_consensus_aura_primitives as consensus_aura;
-
 use rstd::prelude::*;
+use parity_codec_derive::{Encode, Decode};
+use srml_support::{construct_runtime, mashup, mashup_parser, mashup_macro, mashup_macro_impl,
```
Comment (bkchr) [inline node/runtime/src/lib.rs]: Please no^^
I have a commit that fixes that. For the moment, just stick to `macro_use`. I don't want to import all of these.

File: node/runtime/src/lib.rs [hunk 2]
```diff
@@ -20,54 +20,28 @@
...
 use grandpa::fg_primitives::{self, ScheduledChange};
-use client::{
+use substrate_client::impl_runtime_apis;
+use substrate_client::{
 	block_builder::api::{self as block_builder_api, InherentData, CheckInherentsResult},
 	runtime_api as client_api,
 };
-use runtime_primitives::ApplyResult;
+use runtime_primitives::{ApplyResult, create_runtime_str, impl_outer_log, impl_outer_config,
+	__impl_outer_config_types};
```
Comment (bkchr) [inline node/runtime/src/lib.rs]: Here the same for the macros.

File: node/runtime/src/lib.rs (hunk 1 above)
Comment (DarkEld3r) [reply node/runtime/src/lib.rs]: Fixed.

**Comment by gavofyork** [comment]
@bkchr grumbles addressed, i think

**Pull request merged**